### PR TITLE
Start rendering assets using cluster-kube-apiserver operator renderer

### PR DESCRIPTION
**Pull request by mfojtik**
This PR will start generating assets using the new cluster-kube-apiserver-operator image.
Along with manifests and bootstrap static pods that should be used in future by bootkube start, it will provide secrets and configmaps we can feed the openshift-api-operator.

```
[mfojtik@dev-centos generated]$ docker run -v $(pwd)/tls:/assets --rm openshift/origin-cluster-kube-apiserver-operator:latest /usr/bin/cluster-kube-apiserver-operator render --asset-input-dir=/assets --asset-output-dir=/assets/kube-apiserver-bootstrap --config-output-file=/assets/kube-apiserver-bootstrap/config --config-override-file=/usr/share/bootkube/manifests/config/config-overrides.yaml
Writing asset: /assets/kube-apiserver-bootstrap/bootstrap-manifests/kube-apiserver-pod.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/etcd-service.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/openshift-kube-apiserver-ns.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-aggregator-client.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-etcd-client.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-serving-cert.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-client-ca.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-kubelet-serving-ca.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-sa-token-signing-certs.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/kube-apiserver-daemonset.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-kubelet-client.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-aggregator-client-ca.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-etcd-serving-ca.yaml
Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-kube-apiserver-config.yaml
```

/cc @deads2k 
/cc @juanvallejo 
/cc @sttts

### CODE REVIEW

File: modules/bootkube/resources/bootkube.sh [hunk 1]
```diff
@@ -20,6 +20,19 @@ then
 	cp --recursive kco-bootstrap/manifests .
 fi
 
+if [ ! -d kube-apiserver-bootstrap ]
+then
+	echo "Rendering Kubernetes API server core manifests..."
+
+	# shellcheck disable=SC2154
+	podman run \
+		--volume "$PWD:/assets:z" \
```
Comment (mfojtik) [inline modules/bootkube/resources/bootkube.sh]: this assumes the $PWD has the generated/tls secrets, is that assumption correct?

**Comment by mfojtik** [comment]
@deads2k @sttts the command above fail with:

```
F0925 17:06:19.752348       1 render.go:57] failed to generated bootstrap config: failed to load config overrides at "/assets/kube-apiserver-config-overrides.yaml": failed to load "/assets/kube-apiserver-config-overrides.yaml": open /assets/kube-apiserver-config-overrides.yaml: no such file or directory
```

I guess we need `kube-apiserver-config-overrides.yaml` generated and copied by installer to fix this.

**Comment by mfojtik** [comment]
@smarterclayton @deads2k I updated this to fit into new go templates and also added the controller manager render. Once we are confident that the our operators provide same experience as kube-core control plane, we can switch it over just by copying the manifests we need. 

Also this demonstrates that our operator rendering functionality works (we probably should think about the image versions in `bootkube.go` as currently the `:latest` is used.

**Comment by deads2k** [comment]
@abhinavdahiya this is needed to set up the resources used by our operators.  Can you have a look?

/assign @abhinavdahiya

**Comment by abhinavdahiya** [comment]
This PR is more useful
1. if we drop the corresponding things from `tectonic-operators` `kube-core-operator` bump it here
2. use the new rendered assets for bootstrapping.

Otherwise this is unused code-path.
cc @crawford

**Comment by deads2k** [comment]
> This PR is more useful
> 
> 1. if we drop the corresponding things from `tectonic-operators` `kube-core-operator` bump it here
> 2. use the new rendered assets for bootstrapping.
> 
> Otherwise this is unused code-path.
> cc @crawford

We create additional/different resources. Doesn't this start producing those?  We want to enable new things and see them work before removing old.

**Comment by abhinavdahiya** [comment]
> We create additional/different resources. Doesn't this start producing those? We want to enable new things and see them work before removing old.

I meant to say these new files are rendered on disk but not actually used.

**Comment by deads2k** [comment]
> I meant to say these new files are rendered on disk but not actually used.

Where do we put them to have them created?

**Comment by abhinavdahiya** [comment]
@deads2k 
https://github.com/openshift/installer/blob/7f8f39703b99af3d68582a1503884b2498bb4e5d/pkg/asset/ignition/content/bootkube.go#L32-L47

kube-core-operator renders it assets in 3 dirs:
```console
$ ls -l /opt/tectonic/kco-bootstrap/
bootstrap-configs
bootstrap-manifests
manifests
```

* `bootstrap-configs`
This is copied to `/etc/kubernetes/bootstrap-configs`. And is used by bootstrap control plane.

* `bootstrap-manifests`
This is copied to `/opt/tectonic`; this is then used by `bootkube start` as source of `bootstrap control plane` static pods.

* `manifests`
This merged using `cp` with already existing `/opt/tectonic/manifets` dir. This `/opt/tectonic/manifets` is used by `bootkube start` to push manifests in cluster when api is up.

* `/opt/tectonic/tls` and `/opt/tectonic/auth`
These directories have the tls assets and kubeconfig for bootstrap control plane respectively. These are also used by `bootkube start`.

https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/bootkube/bootstrap.go#L28

**Comment by abhinavdahiya** [comment]
@deads2k @mfojtik any progress on this.

File: pkg/asset/ignition/content/bootkube.go [hunk 2]
```diff
@@ -65,7 +68,40 @@ then
...
+	# shellcheck disable=SC2154
+	podman run \
+		--volume "$PWD/tls:/assets:z" \
+		"${KUBE_APISERVER_OPERATOR_IMAGE}" \
+		/usr/bin/cluster-kube-apiserver-operator render \
+		--asset-input-dir=/assets \
+		--asset-output-dir=/assets/kube-apiserver-bootstrap \
+		--config-override-file=/usr/share/bootkube/manifests/config/config-overrides.yaml \
+		--config-output-file=/assets/kube-apiserver-bootstrap/config
+
```
Comment (abhinavdahiya) [inline pkg/asset/ignition/content/bootkube.go]: can you add a FIXME / TODO for using other manifests from this render step.

**Comment by abhinavdahiya** [comment]
if the tests are green; i will `/lgtm`

**Comment by abhinavdahiya** [comment]
@mfojtik 

`journalctl -u bootkube` on bootstrap node.
```console
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Rendering Kubernetes API server core manifests...
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/bootstrap-manifests/kube-apiserver-pod.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-etcd-serving-ca.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-kube-apiserver-config.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/openshift-kube-apiserver-ns.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-aggregator-client.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-kubelet-client.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-aggregator-client-ca.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-client-ca.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/etcd-service.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/kube-apiserver-daemonset.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-etcd-client.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/secret-serving-cert.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-kubelet-serving-ca.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: Writing asset: /assets/kube-apiserver-bootstrap/manifests/configmap-sa-token-signing-certs.yaml
Oct 03 17:42:07 adahiya-0-bootstrap bootkube.sh[2225]: cp: cannot stat ‘kube-apiserver-bootstrap/manifests/secret-*’: No such file or directory
```

File: pkg/asset/ignition/content/bootkube.go [hunk 3]
```diff
@@ -65,7 +68,44 @@ then
...
 fi
 
-if [ ! -d mco-bootstrap ]
+if [ ! -d kube-apiserver-bootstrap ]
+then
+	echo "Rendering Kubernetes API server core manifests..."
+
+	# shellcheck disable=SC2154
+	podman run \
+		--volume "$PWD/tls:/assets:z" \
```
Comment (abhinavdahiya) [inline pkg/asset/ignition/content/bootkube.go]: this should be `--volume "$PWD:/assets:z"` and `--asset-input-dir=/assets/tls`

File: pkg/asset/ignition/content/bootkube.go [hunk 4]
```diff
@@ -65,7 +68,44 @@ then
...
+		--asset-output-dir=/assets/kube-controller-manager-bootstrap \
+		--config-override-file=/usr/share/bootkube/manifests/config/config-overrides.yaml \
+		--config-output-file=/assets/kube-controller-manager-bootstrap/config
+
+	# TODO: copy the bootstrap manifests to replace kube-core-operator
+	cp --recursive kube-apiserver-bootstrap/manifests/secret-* .
+	cp --recursive kube-apiserver-bootstrap/manifests/configmap-* .
+fi
+
+if [ ! -d "mco-bootstrap" ]
```
Comment (wking) [inline pkg/asset/ignition/content/bootkube.go]: nit: no need to add quotes here, none of the characters are sensitive.

**Comment by abhinavdahiya** [comment]
@mfojtik 
Another error:
```
Oct 03 17:53:38 adahiya-0-bootstrap bootkube.sh[1071]: Rendering Kubernetes Controller Manager core manifests...
Oct 03 17:53:41 adahiya-0-bootstrap bootkube.sh[1071]: Trying to pull registry.svc.ci.openshift.org/openshift/origin-v4.0-20181003161809@sha256:b6d1f10f70fef40f4594cc4bfec1b9dad344e9b79710ee29e612dc4f7aec98fe...Getting image source signatures
Oct 03 17:53:42 adahiya-0-bootstrap bootkube.sh[1071]: Skipping fetch of repeat blob sha256:256b176beaff7815db2a93ee2071621ae88f451bb1e198ca73010ed5bba79b65
Oct 03 17:53:42 adahiya-0-bootstrap bootkube.sh[1071]: Copying blob sha256:d486dfcfd9f0b7613cecd5f61574556bd8b1b050cd8608aa943160ea092c2d92
Oct 03 17:53:45 adahiya-0-bootstrap bootkube.sh[1071]: [253B blob data]
Oct 03 17:53:45 adahiya-0-bootstrap bootkube.sh[1071]: Copying config sha256:bfa8856f19cc380aff239779cb6fdb2e3780172329f34739a4108062e5fa4948
Oct 03 17:53:45 adahiya-0-bootstrap bootkube.sh[1071]: [37B blob data]
Oct 03 17:53:45 adahiya-0-bootstrap bootkube.sh[1071]: Writing manifest to image destination
Oct 03 17:53:45 adahiya-0-bootstrap bootkube.sh[1071]: Storing signatures
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: panic: asset: Asset(v3.11.0/kube-apiserver/defaultconfig.yaml): Asset v3.11.0/kube-apiserver/defaultconfig.yaml not found
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: goroutine 1 [running]:
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/v311_00_assets.MustAsset(0xc4200474d0, 0x29, 0x3, 0xc4200474d0, 0x29)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/v311_00_assets/bindata.go:458 +0x12e
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.(*renderOpts).configFromDefaultsPlusOverride(0xc42054d0a0, 0xc42057d540, 0xc420560580, 0x3a, 0x3a, 0xc4200e1228,
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:198 +0x8f
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.(*renderOpts).Run(0xc42054d0a0, 0x0, 0x0)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:162 +0x21d
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.NewRenderCommand.func1(0xc4200f3400, 0xc42032ac00, 0x0, 0x4)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:59 +0x88
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200f3400, 0xc42032ab80, 0x4, 0x4, 0xc4200f3400, 0xc42032ab80)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200f2780, 0xc4200f2780, 0xc4200f3400, 0xc4200f2a00)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:846 +0x30a
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200f2780, 0x16bc200, 0x21ca160)
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:794 +0x2b
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: main.main()
Oct 03 17:53:47 adahiya-0-bootstrap bootkube.sh[1071]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/main.go:30 +0x187
```

**Comment by mfojtik** [comment]
@abhinavdahiya fixed here: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/30

we will need image rebuild :-(

**Comment by abhinavdahiya** [comment]
The tests are green

/lgtm

**Comment by sjenning** [comment]
just puled this down to test and got this
```
Oct 03 19:12:04 dev-bootstrap bootkube.sh[11912]: Found Machine Config Operator's image: registry.svc.ci.openshift.org/openshift/origin-v4.0-20181003181114@sha256:b72e69cd94dc752b1f002383cccf2cc3aa3f591ef3a5bd0060bdba5e065f3419
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: Rendering Kubernetes Controller Manager core manifests...
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: panic: asset: Asset(v3.11.0/kube-apiserver/defaultconfig.yaml): Asset v3.11.0/kube-apiserver/defaultconfig.yaml not found
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: goroutine 1 [running]:
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/v311_00_assets.MustAsset(0xc42038d4d0, 0x29, 0x3, 0xc42038d4d0, 0x29)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/v311_00_assets/bindata.go:458 +0x12e
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.(*renderOpts).configFromDefaultsPlusOverride(0xc4205350a0, 0xc42054f5e0, 0xc420548580, 0x3a, 0x3a, 0xe0, 0x15d3a
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:198 +0x8f
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.(*renderOpts).Run(0xc4205350a0, 0x0, 0x0)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:162 +0x21d
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render.NewRenderCommand.func1(0xc4200d9400, 0xc420348c00, 0x0, 0x4)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/render/render.go:59 +0x88
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).execute(0xc4200d9400, 0xc420348b80, 0x4, 0x4, 0xc4200d9400, 0xc420348b80)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200d8780, 0xc4200d8780, 0xc4200d9400, 0xc4200d8a00)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:846 +0x30a
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200d8780, 0x16bc200, 0x21ca160)
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/vendor/github.com/spf13/cobra/command.go:794 +0x2b
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: main.main()
Oct 03 19:12:06 dev-bootstrap bootkube.sh[11912]: /go/src/github.com/openshift/cluster-kube-controller-manager-operator/cmd/cluster-kube-controller-manager-operator/main.go:30 +0x187
Oct 03 19:12:06 dev-bootstrap systemd[1]: bootkube.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Oct 03 19:12:06 dev-bootstrap systemd[1]: Unit bootkube.service entered failed state.
Oct 03 19:12:06 dev-bootstrap systemd[1]: bootkube.service failed.
```

**Comment by deads2k** [comment]
> just puled this down to test and got this

check to see if your images are new.  The new ones just pushed.

File: pkg/asset/ignition/content/bootkube.go [hunk 5]
```diff
@@ -65,6 +68,45 @@ then
...
+		"${KUBE_CONTROLLER_MANAGER_OPERATOR_IMAGE}" \
+		/usr/bin/cluster-kube-controller-manager-operator render \
+		--asset-input-dir=/assets/tls \
+		--asset-output-dir=/assets/kube-controller-manager-bootstrap \
+		--config-override-file=/usr/share/bootkube/manifests/config/config-overrides.yaml \
+		--config-output-file=/assets/kube-controller-manager-bootstrap/config
+
+	# TODO: copy the bootstrap manifests to replace kube-core-operator
+	cp --recursive kube-apiserver-bootstrap/manifests/openshift-kube-controller-manager-ns.yaml 00_openshift-kube-controller-manager-ns.yaml
+	cp --recursive kube-apiserver-bootstrap/manifests/secret-* .
```
Comment (abhinavdahiya) [inline pkg/asset/ignition/content/bootkube.go]: `cp --recursive kube-apiserver-bootstrap/manifests/secret-* manifests/`

**Comment by sjenning** [comment]
https://origin-release.svc.ci.openshift.org/

last two builds failed CI verification so I don't think https://github.com/openshift/cluster-kube-controller-manager-operator/pull/30 is in the image yet

**Comment by sjenning** [comment]
still having bootkube fail the first time with
```
bootkube.sh[808]: cp: cannot stat ‘kube-apiserver-bootstrap/manifests/openshift-kube-controller-manager-ns.yaml’: No such file or directory
```

File: pkg/asset/ignition/content/bootkube.go [hunk 6]
```diff
@@ -65,6 +68,46 @@ then
...
+		--volume "$PWD:/assets:z" \
+		"${KUBE_CONTROLLER_MANAGER_OPERATOR_IMAGE}" \
+		/usr/bin/cluster-kube-controller-manager-operator render \
+		--asset-input-dir=/assets/tls \
+		--asset-output-dir=/assets/kube-controller-manager-bootstrap \
+		--config-override-file=/usr/share/bootkube/manifests/config/config-overrides.yaml \
+		--config-output-file=/assets/kube-controller-manager-bootstrap/config
+
+	# TODO: copy the bootstrap manifests to replace kube-core-operator
+	cp --recursive kube-apiserver-bootstrap/manifests/openshift-kube-controller-manager-ns.yaml manifests/00_openshift-kube-controller-manager-ns.yaml
```
Comment (abhinavdahiya) [inline pkg/asset/ignition/content/bootkube.go]: `cp --recursive kube-controller-manager-bootstrap/manifests/openshift-kube-controller-manager-ns.yaml manifests/00_openshift-kube-controller-manager-ns.yaml`

**Comment by mfojtik** [comment]
> still having bootkube fail the first time with
> 
> ```
> bootkube.sh[808]: cp: cannot stat ‘kube-apiserver-bootstrap/manifests/openshift-kube-controller-manager-ns.yaml’: No such file or directory
> ```

@sjenning fixed, I hate bash...

@abhinavdahiya can you re-tag please, hopefully last time...

**Comment by wking** [comment]
With multiple folks (including the bot ;) banging away on `/retest`, it's helpful (for me at least) to drop few lines from the error you saw into a comment here (e.g. [here][1]).  That makes it easier to see if we're hitting the same error each time, in which case it's likely to be a real bug and not a temporary flake.

[1]: https://github.com/openshift/installer/pull/396#issuecomment-426485986

**Comment by wking** [comment]
[e2e][1]:

```
error: .status.conditions accessor error: Failure is of the type string, expected map[string]interface{}
timeout waiting for router to be available
2018/10/03 20:56:49 Container test in pod e2e-aws failed, exit code 1, reason Error
2018
```

We've seen that before, e.g. [here][2].  It's [a wait flake][3].

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/322/pull-ci-openshift-installer-master-e2e-aws/414/build-log.txt
[2]: https://github.com/openshift/installer/pull/403#issuecomment-426780782
[3]: https://github.com/openshift/installer/pull/390#issuecomment-426394306

**Comment by mfojtik** [comment]
failure was:

```
Error: Error applying plan:

3 error(s) occurred:

* module.bootstrap.aws_iam_role.bootstrap: 1 error(s) occurred:

* aws_iam_role.bootstrap: Error creating IAM Role ci-op-0gmd4k3x-1d3f3-bootstrap-role: EntityAlreadyExists: Role with name ci-op-0gmd4k3x-1d3f3-bootstrap-role already exists.
	status code: 409, request id: 64ced276-c7af-11e8-b362-6b3d06569f8f
* module.masters.aws_iam_role.master_role: 1 error(s) occurred:

* aws_iam_role.master_role: Error creating IAM Role ci-op-0gmd4k3x-1d3f3-master-role: EntityAlreadyExists: Role with name ci-op-0gmd4k3x-1d3f3-master-role already exists.
	status code: 409, request id: 64ceab1c-c7af-11e8-9915-c37dbaece9fa
* module.iam.aws_iam_role.worker_role: 1 error(s) occurred:

* aws_iam_role.worker_role: Error creating IAM Role ci-op-0gmd4k3x-1d3f3-worker-role: EntityAlreadyExists: Role with name ci-op-0gmd4k3x-1d3f3-worker-role already exists.
	status code: 409, request id: 64ce5d08-c7af-11e8-b78d-41b07c69b921
```

/retest

**Comment by mfojtik** [comment]
failed with:

```
which: no extended.test in (/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin)
/bin/bash: line 93: ginkgo: command not found
```

/retest

**Comment by wking** [comment]
/hold

Waiting on #415 to unstick CI.

**Comment by wking** [comment]
#415 is in.

/hold cancel
/retest

**Pull request merged**